docs/source/examples/example_dates.rst

```diff
@@ -0,0 +1,158 @@
+Example: Dates
+==============
+
+This example concerns itself with expressing ``Constraint`` s against data revolving
+around dates. While date ``Constraint`` s between tables exist, we will only illustrate
+``Constraint`` s on a single table and reference values here. As a consequence, we will
+only use ``WithinRequirement``, as opposed to ``BetweenRequirement``.
+
+Concretely, we will assume a table containing prices for a given product.
+Importantly, these prices are valid for a certain date range only. More precisely,
+we assume that the price is indicated in the ``price`` column, the date from which
+it is valid - the date itself included - in ``date_from`` and the the until when
+it is valid - the date itself included - in the ``date_to`` column.
+
+Such a table might look as follows:
+
+.. list-table:: prices
+   :header-rows: 1
+
+   * - price
+     - date_from
+     - date_to
+   * - 13.99
+     - 22/01/01
+     - 22/01/10
+   * - 14.5
+     - 22/01/11
+     - 22/01/17
+   * - 13.37
+     - 22/01/16
+     - 22/01/31
+
+Given this table, we would like to ensure that 6 constraints are satisfied:
+
+1. All values from column ``date_from`` should be in January 2022.
+2. All values from column ``date_to`` should be in January 2022.
+3. The minimum value in column ``date_from`` should be the first of January 2022.
+4. The maximum value in column ``date_to`` should be the 31st of January 2022.
+5. There is no gap between ``date_from`` and ``date_to``. In other words, every date
+   of January can be assigned to at least one row.
+6. There is no overlap between ``date_from`` and ``date_to``. In other words, every
+   date of January can be assigned to at most one row.
+
+
+Assuming that such a table exists in database, we can write a specification against it.
+
+.. code-block:: python
+
+    import pytest
+    import sqlalchemy as sa
+
+    from datajudge import WithinRequirement
+    from datajudge.pytest_integration import collect_data_tests
+
+    # We create a Requirement, within a table. This object will contain
+    # all the constraints we want to test on the specified table.
+    # To test another table or test the same table against another table,
+    # we would create another Requirement object.
+    prices_req = WithinRequirement.from_table(
+        db_name="example", schema_name=None, table_name="prices"
+    )
+
+    # Constraint 1:
+    # All values from column date_from should be in January 2022.
+    prices_req.add_date_between_constraint(
+        column="date_from",
+	lower_bound="'20220101'",
```

Comment:
Not a discussion for now, but cannot help to see this and have thoughts.

This format for a date works, but shouldn't we be supporting ISO format?

Also, it's strictly a date or can this constraint use datetimes?

Reply:
I created an issue:
https://github.com/Quantco/datajudge/issues/58